hudi-flink/src/main/java/org/apache/hudi/sink/partitioner/GlobalHoodieRecordLocation.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hudi.sink.partitioner;
+
+import org.apache.hudi.common.model.HoodieRecordLocation;
+
+import java.io.Serializable;
+import java.util.Objects;
+
+/**
+ * Similar with {@link org.apache.hudi.common.model.HoodieRecordLocation} but with partition path.
+ */
+public class GlobalHoodieRecordLocation implements Serializable {
```

Comment:
This looks like a `HoodieRecordLocation` + partition path. Can we extend from it and put this to hudi-common module?

Reply:
Should we ? Now only Flink writer needs this, i can move if you think it is necessary.

Reply:
I think that's necessary, maybe other engines will use this later. How about renaming this to `HoodieRecordGlobalLocation`

Reply:
`HoodieGlobalRecordLocation` seems better name.

Reply:
`HoodieRecord` is a noun. IMO the change should be from `Location` to `GlobalLocation`